File: Android/app/src/main/java/com/example/graveapp/api/GraphModel.java
package com.example.graveapp.api;

import com.google.gson.annotations.SerializedName;

import lombok.Getter;

public class GraphModel {

    @SerializedName("edges")
    @Getter
    private Edge[] edges;
    @SerializedName("vertices")
    @Getter
    private Vertex[] vertices;

    public class Edge
    {
        @SerializedName("src")
        @Getter
        private String source;
        @SerializedName("dest")
        @Getter
        private String destination;
        @SerializedName("weight")
        @Getter
        private double weight;
    }

    public class Vertex
    {
        @SerializedName("label")
        @Getter
        private String label;
        @SerializedName("latitude")
        @Getter
        private double latitude;
        @SerializedName("longitude")
        @Getter
        private double longitude;
    }

}

File: graveyardAPI/Models/GraphModel.cs
using System.Text.Json.Serialization;

namespace graveyardAPI.Models
{

    public interface IGraphModel
    {
        Edge[] edges { get; set; }
        Vertex[] vertices { get; set; }
    }

    public class GraphModel : IGraphModel
    {
        [JsonPropertyName("edges")]
        public Edge[] edges { get; set; }
        [JsonPropertyName("vertices")]
        public Vertex[] vertices { get; set; }
    }

    public interface IVertex
    {
        string Label { get; set; }
        double Latitude { get; set; }
        double Longitude { get; set; }
    }

    public class Vertex : IVertex
    {
        [JsonPropertyName("label")]
        public string Label { get; set; }
        [JsonPropertyName("latitude")]
        public double Latitude { get; set; }
        [JsonPropertyName("longitude")]
        public double Longitude { get; set; }
    }

    public interface IEdge
    {
        string Source { get; set; }
        string Destination { get; set; }
        double Weight { get; set; }
    }

    public class Edge : IEdge
    {
        [JsonPropertyName("src")]
        public string Source { get; set; }
        [JsonPropertyName("dest")]
        public string Destination { get; set; }
        [JsonPropertyName("weight")]
        public double Weight { get; set; }
    }

}